logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/EnrollE2EIUseCase.kt

```diff
@@ -77,18 +78,29 @@ class EnrollE2EIUseCaseImpl internal constructor(
 
         prevNonce = newOrderResponse.second
 
-        val authzResponse = e2EIRepository.createAuthz(prevNonce, newOrderResponse.first.authorizations[0]).getOrFail {
+        val firstAuthorizationResponse = e2EIRepository.createAuthorization(prevNonce, newOrderResponse.first.authorizations[0]).getOrFail {
             return E2EIEnrollmentResult.Failed(E2EIEnrollmentResult.E2EIStep.AcmeNewAuthz, it).toEitherLeft()
         }
+        prevNonce = firstAuthorizationResponse.nonce
+        val secondAuthorizationResponse =
+            e2EIRepository.createAuthorization(prevNonce, newOrderResponse.first.authorizations[1]).getOrFail {
+                return E2EIEnrollmentResult.Failed(E2EIEnrollmentResult.E2EIStep.AcmeNewAuthz, it).toEitherLeft()
+            }
+
+        var oidcAuthz = if (firstAuthorizationResponse.challengeType == AuthorizationChallengeType.OIDC)
+            firstAuthorizationResponse else secondAuthorizationResponse
+        var dPopAuthz= if (firstAuthorizationResponse.challengeType == AuthorizationChallengeType.DPoP)
+            firstAuthorizationResponse else secondAuthorizationResponse
```

Comment:
for clarity&consistency can we use the full name instead of acronyms `oidcAuthz` `prevNonce`